trace/example_test.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trace_test
+
+import (
+	"context"
+	"fmt"
+
+	credentials "github.com/oras-project/oras-credentials-go"
+	"github.com/oras-project/oras-credentials-go/trace"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+func Example() {
```

Comment:
Wondering would it be better to make it a [larger example](https://go.dev/blog/examples#larger-examples) 🤔

Reply:
I now think a larger example is better, but currently we can't make that happen as `godoc` works weirdly.